Clarify that CSR API isn't guaranteed to use the cluster CA (#15547)
kubernetes/website: content/en/docs/tasks/tls/managing-tls-in-a-cluster.md

certificate, by the API server to validate kubelet client certificates, etc. To
support this, the CA certificate bundle is distributed to every node in the
cluster and is distributed as a secret attached to default service accounts.

> Optionally, your workloads can use this CA to establish trust. Your application

Reviewer: The CA distributed to pods in the service account token volume is *only* guaranteed to be usable for validating the kube-apiserver's serving certificate, and should not be used for any other purpose. This paragraph (and this entire doc) is misleading on that point.

It is useful to describe how to request client and serving certificates from the CSR API, but the instructions should *explicitly* state that the associated trust bundle is not distributed via the API or logically provided to pods, and that cluster admins are currently responsible for distribution to:
* things that need to verify the client certs (typically kube-apiserver and kubelet verification of client certs with `--client-ca-file`)
* things that need to verify serving certs (typically kube-apiserver verification of kubelet serving certs with `--kubelet-certificate-authority`)

Author: Reworded everything to stop implying cluster CA signs certs.
PTAL

can request a certificate signing using the `certificates.k8s.io` API using a
protocol that is similar to the
[ACME draft](https://github.com/ietf-wg-acme/acme/).

> Note: Certificates signed using the `certificates.k8s.io` API are not

Reviewer 2: I recommend using a `{{< note >}}` Hugo shortcode here.

guaranteed to be signed by the cluster root CA. The CA used for signing depends

> on the cluster setup. CA certificate bundle distributed to nodes and service

Reviewer 2: If possible, reword this to use the active voice and to [address the reader as “you”](https://kubernetes.io/docs/contribute/style/style-guide/#address-the-reader-as-you).

> accounts may not contain the CA use to sign certificates using the

Reviewer 2: ```suggestion
accounts might not contain the CA used to sign certificates using the
```

certificate chain and adding the parsed certificates to the `RootCAs` field
in the [`tls.Config`](https://godoc.org/crypto/tls#Config) struct.

The cluster root CA certificate bundle is automatically mounted into pods using
the default service account at the path
`/var/run/secrets/kubernetes.io/serviceaccount/ca.crt`.  If you are not using

> the default service account, ask a cluster administrator to build a configmap

Reviewer 2: Does this help a reader who is their own cluster operator? If not, is there a way to signpost that reader to learn about making that ConfigMap?

(If there's no existing document then, IMO, it's OK to track that in a new issue and leave things there).
